pydantic/_internal/_generate_schema.py

```diff
@@ -171,6 +171,8 @@ def apply_each_item_validators(
     field_name: str | None,
 ) -> core_schema.CoreSchema:
     # This V1 compatibility shim should eventually be removed
+    if not each_item_validators:
+        return schema
```

Comment:
```suggestion

    # fail early if each_item_validators is empty
    if not each_item_validators:
        return schema
```

Reply:
I felt like this needed some clarification / space from the above comment, which referred to the function as a whole, but became ambiguous with this addition.